src/bootstrap/lib.rs

```diff
@@ -668,6 +668,7 @@ impl Build {
             job::setup(self);
         }
 
+        let _ = &builder::Builder::new(&self).initial_rustfmt();
```

Comment:
This looks good, thanks :) can you just add a comment explaining what's going on?
```suggestion
        // Download rustfmt early so that it can be used in rust-analyzer configs.
        let _ = &builder::Builder::new(&self).initial_rustfmt();
```